realm/realm-library/src/main/java/io/realm/internal/modules/CompositeMediator.java

```diff
@@ -106,6 +106,9 @@ public void insert(Realm realm, RealmModel object, Map<RealmModel, Long> cache)
 
     @Override
     public void insert(Realm realm, Collection<? extends RealmModel> objects) {
+        if (objects.isEmpty()) {
```

Comment:
IMO we should move these checks to the `Realm` class instead as that is where the `null` check already is. I would avoid spreading them out across multiple classes.